cmd/geth/main.go

```diff
@@ -398,3 +391,30 @@ func startNode(ctx *cli.Context, stack *node.Node) {
 		}
 	}
 }
+
+// unlockAccounts unlocks any account specifically requested.
+func unlockAccounts(ctx *cli.Context, stack *node.Node) {
+	// personalExposed checks whether personal API is exposed by http.
+	personalExposed := func() bool {
+		for _, module := range stack.Config().HTTPModules {
```

Comment:
We should not tie it to `personal` rather to the entire RPC (both HTTP and WS). It's safer as `eth` is also susceptible.